### ISSUE TITLE
Updating tags attribute on object fails

### ISSUE THREAD

**Issue by jord-bh** [digitalocean/pynetbox#94]
Hi,

I'm currently testing around with tags and tried to append a tag to my interface. When I call the save() function, it doesn't seem to properly register any changes and as such does not perform the update. If I update _another_ field such as the description, the change is however stored. 

Hopefully the following code illustrates the issue:

```
>>> import pynetbox
>>> nb = pynetbox.api(url='https://sub.domain.tld/', token='0123456789abcdef0123456789abcdef01234567')
>>> intf = nb.dcim.interfaces.get(name='eth0')
>>> intf.tags
[]
>>> intf.tags.append('27')
>>> intf.save()
False
>>> nb.dcim.interfaces.get(name='eth0').tags
[]
>>> intf.description = 'test'
>>> intf.save()
True
>>> nb.dcim.interfaces.get(name='eth0').tags
['27']
```

I just tested tagging a device and met the same problem there, it seems like the issue is related to the tags attribute rather than interface object being incorrectly handled.

**Comment by tfuzeau** [comment]
Hello,

I got the same issue and dig a bit why it isn't working. I finally found this is due to the __compare()_ which is returning **True**, so the real save didn't apply.

I didn't look for the exact issue inside the __compare()_, but I create a quick workaround by "force" saving the change:

```
from pynetbox.lib.query import Request
def force_save(obj):
    if obj.save():
        return True
    req = Request(
        key=obj.id,
        base=obj.endpoint_meta.get('url'),
        token=obj.api_kwargs.get('token'),
        session_key=obj.api_kwargs.get('session_key'),
        version=obj.api_kwargs.get('version'),
        ssl_verify=obj.api_kwargs.get('ssl_verify')
    )
    return req.put(obj.serialize())
```

**Comment by zachmoody** [comment]
Had a brief opportunity to dig into this. It was a few days ago, so I might be off on some details. You're right, appending does cause `_compare()` to miss the changes. However, passing a complete list plus the item you're wanting to add works. This is the part I've been stuck on. We can make append work *but* using it breaks idempotence. The tag field behaves like a `set()`; I'm just wondering if it should be exposed to the user as one or just converted to one somewhere in the serialization process. :thinking:

**Comment by jord-bh** [comment]
Hey @zachmoody, I'm unsure if I'm doing something wrong but I tried the exact same procedure after upgrading pynetbox and I'm seeing the same behaviour:

```
>>> import pynetbox
>>> pynetbox.__version__
'3.4.9'
>>> nb = pynetbox.api(url='https://sub.domain.tld/', token='0123456789abcdef0123456789abcdef01234567')
>>> intf = nb.dcim.interfaces.get(name='eth0')
>>> intf.tags
[]
>>> intf.tags.append('27')
>>> intf.save()
False
>>> nb.dcim.interfaces.get(name='eth0').tags
[]
>>> intf.description = 'test'
>>> intf.save()
True
>>> nb.dcim.interfaces.get(name='eth0').tags
['27']
```

**Comment by zachmoody** [comment]
:disappointed: Sorry about that, only part of the fix got ported over from the 4.0 branch. I'll see about getting the rest of it in a new 3.x release today.

**Comment by jord-bh** [comment]
Thank you!
